pkg/scheduler/objects/application.go

```diff
@@ -1015,8 +1072,11 @@ func (sa *Application) GetAllAllocations() []*Allocation {
 
 // get a copy of all placeholder allocations of the application
 // No locking must be called while holding the lock
-func (sa *Application) getPlaceholderAllocations() []*Allocation {
+func (sa *Application) GetPlaceholderAllocations() []*Allocation {
 	var allocations []*Allocation
+	if sa == nil || len(sa.allocations) == 0 {
+		return allocations
+	}
```

Comment:
We don't need to export it. It remained exported because in my first implementation I did the cleanup from the partition mananger. I will make it private